testutils/clustermanager/gke.go

```diff
@@ -179,42 +154,51 @@ func (gc *GKECluster) Provider() string {
 // in us-central1, and default BackupRegions are us-west1 and us-east1. If
 // Region or Zone is provided then there is no retries
 func (gc *GKECluster) Acquire() error {
-	if err := gc.initialize(); err != nil {
-		return fmt.Errorf("failed initializing with environment: '%v'", err)
+	if err := gc.checkEnvironment(); err != nil {
+		return fmt.Errorf("failed checking project/cluster from environment: '%v'", err)
 	}
-	gc.ensureProtected()
-	var err error
-	// Check if using existing cluster
+	// If gc.Cluster is discovered above, then the cluster exists and it's
+	// project and name matches with requested, use it
 	if gc.Cluster != nil {
+		gc.ensureProtected()
 		return nil
 	}
 	if gc.Request.SkipCreation {
-		log.Println("Skipping cluster creation as SkipCreation is set")
-		return nil
+		return errors.New("cannot acquire cluster if SkipCreation is set")
 	}
 
-	// Make a deep copy of the request struct, since the original request is supposed to be immutable
-	request := gc.Request.DeepCopy()
-	// Perform GKE specific cluster creation logics
-	if request.ClusterName == "" {
-		request.ClusterName, err = getResourceName(ClusterResource)
+	// If comes here we are very likely going to create a cluster, unless
+	// the cluster already exists
+
+	// Cleanup if cluster is created by this client
+	gc.NeedsCleanup = !common.IsProw()
+
+	// Get project name from boskos if running in Prow, otherwise it should fail
+	// since we don't know which project to use
+	if common.IsProw() {
+		project, err := gc.boskosOps.AcquireGKEProject(nil)
 		if err != nil {
-			return fmt.Errorf("failed getting cluster name: '%v'", err)
+			return fmt.Errorf("failed acquiring boskos project: '%v'", err)
 		}
+		gc.Project = &project.Name
 	}
-	if request.Project == "" {
-		request.Project = *gc.Project
+	if gc.Project == nil {
+		return errors.New("gcp project must be set")
```

Comment:
```suggestion
		return errors.New("GCP project must be set")
```
Capitalized names can still be capitalized in the errors :-)